lib/validators.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const Joi = require('joi');
+
+const errors = require('./errors');
+
+const validators = {schemas: {}};
+
+validators.schemas.requestData = Joi.object().keys({
+  clientId: Joi.string().guid({
+    version: [
+      'uuidv4',
+    ],
+  }).required(),
+  clientSecret: Joi.string().guid({
+    version: [
+      'uuidv4',
+    ],
+  }).required(),
+  redirectUri: Joi.string().uri().required(),
+  scope: Joi.array().items(Joi.string()),
+  development: Joi.boolean(),
+});
+
+validators.validateRequestData = function(req) {
+  try {
+    Joi.assert(req, validators.schemas.requestData);
+  } catch (err) {
+    throw new errors.ValidationError(err);
```

Comment:
why not just surface the `Joi` error here?

Reply:
Shouldn't make much of a difference because Smartcar error picks it up. Just allows extensibility in the future to throw/deal with the errors more easily. We seem to be throwing errors through ours (Smartcar error), so it's also consistent with our error throwing. Thoughts?

Reply:
The difference it does make is discoverability, the Joi error contains a lot of information that is helpful in determining what exactly was invalid and wrapping it would hide that information one level deeper, at some level `console.log` stops to recurse and just says `[Object]` but idk if one level of wrapping would hit that limit

another reason I wouldn't want to wrap it is that I think `SmartcarError` should be just for API errors, this is more of a `TypeError` if we were using a typed language as opposed to a validation errors returned by our api